### PR TITLE
DET-564: JS SDK

### PR DESCRIPTION
**Pull request by stephanwampouille**
Parallel ticket for https://github.com/preludeorg/libraries/pull/799 for JS as well as minor consistency changes for python sdk

### CODE REVIEW

File: javascript/sdk/lib/controllers/iam.ts [hunk 1]
```diff
@@ -76,16 +76,38 @@ export default class IAMController {
 
   /** Create a new user inside an account */
   async createUser(
```
Comment (kenrick) [inline javascript/sdk/lib/controllers/iam.ts:78]: This should be a new function called something like `verifyUser` or `verifyToken` which does a GET to `/iam/user` and passes a token to `?token={token}`. This should use `this.#client.request("/iam/user"` so we can call it without being logged in

**Review by kenrick — CHANGES_REQUESTED** [review]
After speaking with Alex noticed these changes.

File: javascript/sdk/lib/types.ts [hunk 2]
```diff
@@ -67,6 +67,7 @@ export interface CreateUserParams {
   email: string;
   name?: string;
   expires?: string;
+  token?: string;
```
Comment (kenrick) [inline javascript/sdk/lib/types.ts:70]: You can remove this as well

**Review by kenrick — CHANGES_REQUESTED** [review]

File: javascript/sdk/lib/controllers/iam.ts (hunk 1 above)
Comment (stephanwampouille) [reply javascript/sdk/lib/controllers/iam.ts:78]: so what else needs to be in the new function? Just token and thats all?

Comment (stephanwampouille) [reply javascript/sdk/lib/controllers/iam.ts:78]: @khyberspache

File: javascript/sdk/lib/types.ts [hunk 3]
```diff
@@ -56,6 +56,12 @@ export interface Account {
   company: string;
 }
 
+export interface VerifiedUser {
+  account_id: string;
+  token: string;
+  user: User;
```
Comment (khyberspache) [inline javascript/sdk/lib/types.ts:62]: It's the flat properties of the user, not the user object itself

File: python/sdk/prelude_sdk/controllers/iam_controller.py [hunk 4]
```diff
@@ -108,11 +108,11 @@ def delete_user(self, handle):
             return res.json()
         raise Exception(res.text)
 
-    def reset_password(self, handle: str, account_id: str = None):
+    def reset_password(self, email: str, account_id: str = None):
```
Comment (khyberspache) [inline python/sdk/prelude_sdk/controllers/iam_controller.py:111]: Make sure you change the call in the CLI too

**Review by khyberspache — CHANGES_REQUESTED** [review]
2 small items

File: python/sdk/prelude_sdk/controllers/iam_controller.py (hunk 4 above)
Comment (stephanwampouille) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:111]: its actually already email,
Screenshot 2023-08-29 at 6 58 29 PM

File: javascript/sdk/lib/types.ts (hunk 3 above)
Comment (stephanwampouille) [reply javascript/sdk/lib/types.ts]: resolved

File: javascript/sdk/lib/controllers/iam.ts [hunk 5]
```diff
@@ -76,16 +76,38 @@ export default class IAMController {
...
+    return (await response.json()) as CreatedUser;
+  }
+
+  /** Reset a user inside an account */
+  async resetUser(
+    account_id: string,
+    email: string,
+    options: RequestOptions = {}
+  ): Promise<StatusResponse> {
+    const response = await this.#client.requestWithAuth("/iam/user/reset", {
```
[the remark this answers is not in the document]
Comment (stephanwampouille) [reply javascript/sdk/lib/controllers/iam.ts]: resolved

**Comment by kenrick** [comment]
LGTM

**Review by kenrick — APPROVED** [review]

File: python/sdk/prelude_sdk/controllers/iam_controller.py (hunk 4 above)
Comment (khyberspache) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:111]: Handle=email

Change to

Email=email

Comment (stephanwampouille) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:111]: oh shoot lol

Comment (stephanwampouille) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:111]: fixed

**Review by khyberspache — APPROVED** [review]

**Comment by khyberspache** [comment]
Merge when ready

**Review by privateducky — APPROVED** [review]

**Pull request merged**